app/parse.py

```diff
@@ -15,8 +22,121 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def get_price_from_cards(cards: list[WebElement]) -> list[float]:
+    return [float(card.find_element(
+        By.CSS_SELECTOR,
+        "h4.price"
+    ).text.replace("$", ""))
+        for card in cards]
+
+
+def get_rating_from_cards(cards: list[WebElement]) -> list[int]:
+    return [len(card.find_elements(
+            By.CSS_SELECTOR,
+            "div.ratings > p:nth-child(2) > span"
+            ))
+            for card in cards]
+
+
+def get_reviews_from_cards(cards: list[WebElement]) -> list[int]:
+    return [int(card.find_element(
+        By.CSS_SELECTOR,
+        "div.ratings > p.float-end.review-count").text.split()[0])
+        for card in cards]
+
+
+def get_description_from_cards(cards: list[WebElement]) -> list[str]:
+    return [card.find_element(By.CSS_SELECTOR, "div.caption > p").text for
+            card in cards]
+
+
+def get_title_from_cards(cards: list[WebElement]) -> list[str]:
+    list_of_title = []
+    for card in cards:
+        href = card.find_element(
+            By.CSS_SELECTOR,
+            "div.caption > h4:nth-child(2) > a"
+        ).get_attribute("href")
+        req = requests.get(href, verify=False).content
+        soup = BeautifulSoup(req, "html.parser")
+        list_of_title.append(soup.select_one(
+            "div.col-xl-10 > div.caption > h4.card-title"
+        ).text)
+    return list_of_title
+
+
+def get_cards(driver: WebDriver, url: str) -> list[WebElement]:
+    driver.get(url)
+    try:
+        button = driver.find_element(
+            By.CLASS_NAME,
+            "ecomerce-items-scroll-more"
+        )
+    except Exception:
+        return driver.find_elements(By.CLASS_NAME, "card-body")
+    while button.is_displayed():
+        driver.execute_script("arguments[0].click();", button)
+        button = driver.find_element(
+            By.CLASS_NAME,
+            "ecomerce-items-scroll-more"
+        )
+    return driver.find_elements(By.CLASS_NAME, "card-body")
+
+
+def scrap(driver: WebDriver, file_url: str, url: str) -> None:
+    cards = get_cards(driver, url)
+    titles = get_title_from_cards(cards)
+    descriptions = get_description_from_cards(cards)
+    prices = get_price_from_cards(cards)
+    ratings = get_rating_from_cards(cards)
+    reviews = get_reviews_from_cards(cards)
+    products = [
+        Product(title=titles[i],
+                description=descriptions[i],
+                price=prices[i], rating=ratings[i],
+                num_of_reviews=reviews[i]) for i in range(len(cards))]
```

Comment:
```suggestion
        Product(
            title=titles[i],
            description=descriptions[i],
            price=prices[i], 
            rating=ratings[i],
            num_of_reviews=reviews[i]
        ) 
        for i in range(len(cards))
   ]
```
Please fix formation in all places